test/catalog_test.exs

```diff
@@ -34,6 +34,21 @@ defmodule Wanda.CatalogTest do
       end)
     end
 
+    test "should return the whole catalog but the check with the when clause" do
```

Comment:
To have a more meaningful test name:
`should filter out checks if the when clause doesn't match` or similar